agent/agent_method_script.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+DATALINK="$(svcprop -c -p config/datalink "${SMF_FMRI}")"
+GATEWAY="$(svcprop -c -p config/gateway "${SMF_FMRI}")"
+DATASET="$(svcprop -c -p config/dataset "${SMF_FMRI}")"
+LISTEN_ADDR="$(svcprop -c -p config/listen_addr "${SMF_FMRI}")"
+LISTEN_PORT="$(svcprop -c -p config/listen_port "${SMF_FMRI}")"
+PORTBASE="$(svcprop -c -p config/portbase "${SMF_FMRI}")"
+DOWNSTAIRS_PREFIX="$(svcprop -c -p config/downstairs_prefix "${SMF_FMRI}")"
+SNAPSHOT_PREFIX="$(svcprop -c -p config/snapshot_prefix "${SMF_FMRI}")"
+
+ipadm show-addr "$DATALINK/ll" || ipadm create-addr -t -T addrconf "$DATALINK/ll"
+ipadm show-addr "$DATALINK/omicron6"  || ipadm create-addr -t -T static -a "$LISTEN_ADDR" "$DATALINK/omicron6"
+route get -inet6 default -inet6 "$GATEWAY" || route add -inet6 default -inet6 "$GATEWAY"
```

Comment:
We should really move this network initialisation into a separate service that's uniform in all of the zones, I think.  If we're happy with the shape of this (i.e., this sequence of commands with this set of input variables) we could even include it in the _brand_ as effectively a stable interface.  You'd be able to just turn it on and specify the datalink/address/gateway in the site profile and that'd be it.  We could make `svc:/milestone/network` depend on it, and all our services could just depend on _that_.

We can also do that later, of course, if you want to get moving now.

Reply:
I'm totally on-board with doing this , and agree that it would be better served as a service. Is https://github.com/oxidecomputer/helios-omicron-brand a better spot for this? Where should it live in there?

(I do plan on moving forward with this implemented manually, but I'd be happy to switch over to another service if I could get some guidance on setting up the service within the brand)